frontend/src/App.tsx

```diff
@@ -624,6 +626,7 @@ export class App extends PureComponent<Props, State> {
    */
   closeDialog = (): void => {
     this.setState({ dialog: undefined })
+    this.props.s4aCommunication.onModalReset()
```

Comment:
nit: could add a unit test for `closeDialog` (I don't think one exists currently) that checks that `s4aCommunication.onModalReset()` gets called